ablySpec/RealtimeClient.connection.swift

```diff
@@ -1080,6 +1080,106 @@ class RealtimeClientConnection: QuickSpec {
                 }
             }
 
+            // RTN14b
+            pending("connection request fails") {
+
+                // NOTE: the connection doesn't retry to request a new token if any failure occurs
+
+                it("should not emit error with a renewable token") {
+                    let options = AblyTests.commonAppSetup()
+                    options.autoConnect = false
+                    let tokenTtl = 1.0
+                    options.token = getTestToken(key: options.key, ttl: tokenTtl)
+
+                    let client = ARTRealtime(options: options)
+                    client.setTransportClass(TestProxyTransport.self)
+                    defer { client.close() }
+
+                    // Let the token expire
+                    waitUntil(timeout: testTimeout) { done in
+                        delay(tokenTtl) {
+                            done()
+                        }
+                    }
+
+                    waitUntil(timeout: testTimeout) { done in
+                        client.eventEmitter.on { state, errorInfo in
+                            switch state {
+                            case .Connected:
+                                expect(errorInfo).to(beNil())
+                                // New token
+                                expect(client.auth().tokenDetails!.token).toNot(equal(options.token))
+                                done()
+                            case .Failed:
+                                fail("Should not emit error (\(errorInfo))")
+                                done()
+                            default:
+                                break
+                            }
+                        }
+                        client.connect()
+                    }
+
+                    let transport = client.transport as! TestProxyTransport
+                    let failures = transport.protocolMessagesReceived.filter({ $0.action == .Error })
+
+                    if failures.count != 1 {
+                        fail("Should have only one connection request fail")
+                        return
+                    }
+
+                    expect(failures[0].error!.code).to(equal(40142)) //Token expired
+                }
+
+                it("should fail again due a token request error") {
```

Comment:
This should say "it should transition to the Failed state because the token is invalid and not renewable"